kodilla-testing/src/test/java/com/kodilla/testing/library/BookDirectoryTestSuite.java

```diff
@@ -0,0 +1,160 @@
+
+        // Then
+        assertEquals(4, theListOfBooks.size());                                          // [15]
+    }
+
+    @Nested
+    @DisplayName("Books With Condition Test")
+    class testListBookWithCondition {
+        @Test
+        void testListBooksWithConditionMoreThan20() {
+
+            // Given
+            BookLibrary bookLibrary = new BookLibrary(libraryDatabaseMock);                  // [1]
+            List<Book> resultListOf0Books = new ArrayList<Book>();                           // [2]
+            List<Book> resultListOf15Books = generateListOfNBooks(15);                       // [3]
+            List<Book> resultListOf40Books = generateListOfNBooks(40);                       // [4]
+            when(libraryDatabaseMock.listBooksWithCondition(Mockito.anyString()))                    // [5]
+                    .thenReturn(resultListOf15Books);                                             // [6]
+            when(libraryDatabaseMock.listBooksWithCondition("ZeroBooks"))                    // [7]
+                    .thenReturn(resultListOf0Books);                                              // [8]
+            when(libraryDatabaseMock.listBooksWithCondition("FortyBooks"))                   // [9]
+                    .thenReturn(resultListOf40Books);                                             // [10]
+
+            // When
+            List<Book> theListOfBooks0 = bookLibrary.listBooksWithCondition("ZeroBooks");    // [11]
+            List<Book> theListOfBooks15 = bookLibrary.listBooksWithCondition("Any title");   // [12]
+            List<Book> theListOfBooks40 = bookLibrary.listBooksWithCondition("FortyBooks");  // [13]
+            // Then
+
+            assertEquals(0, theListOfBooks0.size());                                         // [14]
+            assertEquals(15, theListOfBooks15.size());                                       // [15]
+            assertEquals(0, theListOfBooks40.size());                                        // [16]
+        }
+
+        @Test
+        void testListBooksWithConditionFragmentShorterThan3() {                          // [1]
+            // Given
+            LibraryDatabase libraryDatabaseMock = mock(LibraryDatabase.class);            // [2]
+            BookLibrary bookLibrary = new BookLibrary(libraryDatabaseMock);               // [3]
+
+            // When
+            List<Book> theListOfBooks10 = bookLibrary.listBooksWithCondition("An");       // [4]
+
+            // Then
+            assertEquals(0, theListOfBooks10.size());                                     // [5]
+            verify(libraryDatabaseMock, times(0)).listBooksWithCondition(anyString());    // [6]
+        }
+    }
+
+    @Nested
+    @DisplayName("BookInHandsOf Test")
+    class testListBookInHandsOf {
+        @Test
+        void testListBookInHandsOf0() {
+            //Given
+            BookLibrary bookLibrary = new BookLibrary(libraryDatabaseMock);
```

Comment:
to jest linia ktora sie powtarza w wielu testach. Wyciagnij to do zmiennej klasowej + tworzenie w @BeforeEach